combine/combine_test.go

```diff
@@ -0,0 +1,267 @@
+			So(err, ShouldBeNil)
+
+			b, err := os.ReadFile(outputPath)
+			So(err, ShouldBeNil)
+			So(string(b), ShouldNotEqual, expectedMergeOutput)
+
+			actualFileContents, err := fs.ReadCompressedFile(outputPath)
+			So(err, ShouldBeNil)
+			So(actualFileContents, ShouldEqual, expectedMergeOutput)
+		})
+
+		Convey("You can merge the file contents to an output", func() {
+			inputFiles := make([]string, len(inputs))
+			for i, file := range inputs {
+				inputFiles[i] = file.Name()
+			}
+
+			sortMergeOutput, cleanup, err := MergeSortedFiles(inputFiles)
+			So(err, ShouldBeNil)
+
+			err = MergeSummaryLines(sortMergeOutput, 3, 2, lineMerger, output)
+			So(err, ShouldBeNil)
+
+			err = cleanup()
+			So(err, ShouldBeNil)
+
+			b, err := os.ReadFile(outputPath)
+			So(err, ShouldBeNil)
+
+			So(string(b), ShouldEqual, expectedMergeLinesOutput)
+		})
+
+		Convey("can merge the file contents to a compressed output", func() {
+			inputFiles := make([]string, len(inputs))
+			for i, file := range inputs {
+				inputFiles[i] = file.Name()
+			}
+
+			sortMergeOutput, cleanup, err := MergeSortedFiles(inputFiles)
+			So(err, ShouldBeNil)
+
+			zw, closeOutput, err := Compress(output)
+			So(err, ShouldBeNil)
+
+			err = MergeSummaryLines(sortMergeOutput, 3, 2, lineMerger, zw)
+			So(err, ShouldBeNil)
+
+			closeOutput()
+
+			err = cleanup()
+			So(err, ShouldBeNil)
+
+			b, err := os.ReadFile(outputPath)
+			So(err, ShouldBeNil)
+			So(b, ShouldNotEqual, expectedMergeLinesOutput)
+
+			actualFileContents, err := fs.ReadCompressedFile(outputPath)
+			So(err, ShouldBeNil)
+			So(actualFileContents, ShouldEqual, expectedMergeLinesOutput)
+		})
+	})
+}
+
+// concatMerger specifies how a set of files is merged to an output.
+func concatMerger(data io.ReadCloser, output io.Writer) error {
+	if _, err := io.Copy(output, data); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// lineMerger is a matchingSummaryLineMerger that, given cols 2, will sum
+// the second to last element of a and b and store the result in a[penultimate],
+// and likewise for the last element in a[last]. This corresponds to summing the
+// file count and size columns of 2 lines in a by* file.
+func lineMerger(cols int, a, b []string) {
+	last := len(a) - (cols - 1)
+	penultimate := last - 1
+
+	a[penultimate] = addNumberStrings(a[penultimate], b[penultimate])
+	a[last] = addNumberStrings(a[last], b[last])
+}
+
+// createInputsAndOutput creates the combine files needed for testing, returning
+// a list of the open files, the open output file and the output file path.
+func createInputsAndOutput(t *testing.T) ([]*os.File, *os.File, string) {
+	dir := t.TempDir()
+
+	paths := [6]string{"path1", "path2", "path3", "path4", "path5", "path6"}
+	users := [3]string{"adam", "ben", "charlie"}
+	groups := [3]string{"david", "fred", "graham"}
+	dirs := [2]string{"short/test/dir/", "test/dir/that/is/much/longer"}
+
+	inputs := make([]*os.File, len(paths))
+
+	for i, path := range paths {
```

Comment:
De-duplicate this code as per the linter. Can refactor the duplicated code out to a new function.